wordpress-mobile/WordPress-iOS
# Removed the old domains list view controller that was no longer used

This PR simply removes an old domains list implementation that appears to have been unused since 2016.

**To test**

* Check the app builds and runs
* Ensure you can still navigate to the new domains area

## Regression Notes

1. Potential unintended areas of impact

None, I searched the rest of the source as well as git history to check that the VC is no longer used.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
